hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/RunJar.java

```diff
@@ -315,6 +316,11 @@ public void run() {
     Thread.currentThread().setContextClassLoader(loader);
     Class<?> mainClass = Class.forName(mainClassName, true, loader);
     Method main = mainClass.getMethod("main", String[].class);
+    
+    if (!Modifier.isStatic(main.getModifiers())) {
+      throw new IOException("Method main must be static: " + mainClassName);
```

Comment:
Can we add a unit test to make sure we trigger this instead of NPE?

Reply:
I've added a test, but I couldn't get Hadoop to run locally—seeing this error:

```
Plugin org.apache.hadoop:hadoop-maven-plugins:3.3.0-SNAPSHOT or one of its dependencies could not be resolved: Could not find artifact org.apache.hadoop:hadoop-maven-plugins:jar:3.3.0-SNAPSHOT
```

Guidance from the [Dev Environment Setup](https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute#HowToContribute-DevEnvironmentSetup) was no help unfortunately. The suggested docker script also fails to build.